### PR TITLE
New Tesseract API without GenericVector and STRING

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

File: include/tesseract/serialis.h [hunk 1]
```diff
@@ -43,14 +41,17 @@ constexpr size_t countof(T const (&)[N]) noexcept {
   return N;
 }
 
-// Function to read a GenericVector<char> from a whole file.
+// Function to read a std::vector<char> from a whole file.
```
Comment (egorpugin) [inline include/tesseract/serialis.h:44]: Probably we should replace `vector<char>` with `vector<uint8_t>` for binary data later.

**Comment by egorpugin** [comment]
Let's wait for builders and merge if they are green.

File: src/api/tesseractmain.cpp [hunk 2]
```diff
@@ -299,12 +299,11 @@ static void SetVariablesFromCLArgs(tesseract::TessBaseAPI* api, int argc,
 }
 
 static void PrintLangsList(tesseract::TessBaseAPI* api) {
-  GenericVector<STRING> languages;
+  std::vector<std::string> languages;
   api->GetAvailableLanguagesAsVector(&languages);
   printf("List of available languages (%d):\n", languages.size());
```
Comment (stweil) [inline src/api/tesseractmain.cpp:304]:
```suggestion
  printf("List of available languages (%zu):\n", languages.size());
```

Comment (stweil) [reply src/api/tesseractmain.cpp:304]: This should fix the issue reported by LGTM. I don't add that now to allow the continuous integration to finish.

**Comment by egorpugin** [comment]
I wonder if we see similar changes but "without `const char *`". :)

**Comment by amitdo** [comment]
Lots of presents in the last few days :-)

@egorpugin and @stweil, thank you!

**Comment by stweil** [comment]
@amitdo, wait with thanks until it works again. The good news is that it can be built, so it is possible to run tests.

**Comment by amitdo** [comment]
std::list vs. std::vector:

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#slcon2-prefer-using-stl-vector-by-default-unless-you-have-a-reason-to-use-a-different-container

If you still think list is better in this particular case, than it's fine.

**Comment by egorpugin** [comment]
I also think that std::list place is not really a hot path (is it?) and does not process gigs of data in vector/list, so it can be just vector.

**Comment by amitdo** [comment]
list->vector
reverted in deec8ef46fbf2f

**Comment by stweil** [comment]
Yes, the code no longer uses `std::list`. The new implementation should also be faster than the old one based on `GenericVector` which might improve the performance for large images with a complex layout.

**Comment by egorpugin** [comment]
I assure you, when we replace more places with modern C++, performance will increase. :)

**Comment by amitdo** [comment]
It seems we moved to C++17 as minimum requirement.

**Comment by amitdo** [comment]
@egorpugin

More places to modernize:
clst, elst, elst2.

Macros can be replaced with templates.

**Pull request merged**